src/agent/src/image_rpc.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2021 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::fs;
+use std::process::{ExitStatus, Command};
+use std::sync::Arc;
+
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use protocols::image;
+use tokio::sync::Mutex;
+use ttrpc::{self, error::get_rpc_status as ttrpc_error};
+
+use crate::sandbox::Sandbox;
+//use crate::trace_rpc_call;
+use crate::rpc::CONTAINER_BASE;
+
+const SKOPEO_PATH: &str = "/usr/bin/skopeo";
+const UMOCI_PATH: &str = "/usr/local/bin/umoci";
+
+// Convenience macro to obtain the scope logger
+macro_rules! sl {
+    () => {
+        slog_scope::logger()
+    };
+}
+
+pub struct ImageService {
+    sandbox: Arc<Mutex<Sandbox>>,
+}
+
+impl ImageService {
+    pub fn new(sandbox: Arc<Mutex<Sandbox>>) -> Self {
+        Self { sandbox }
+    }
+
+    fn pull_image_from_registry(image: &str, cid: &str, source_creds: &Option<&str>) -> Result<()> {
+        let source_image = format!("{}{}", "docker://", image);
+
+        let manifest_path = format!("/tmp/{}/image_manifest", cid);
+        let target_path_manifest = format!("dir://{}", manifest_path);
+
+        // Define the target transport and path for the OCI image, without signature
+        let oci_path = format!("/tmp/{}/image_oci:latest", cid);
+        let target_path_oci = format!("oci://{}", oci_path);
+
+        fs::create_dir_all(&manifest_path)?;
+        fs::create_dir_all(&oci_path)?;
+
+        info!(sl!(), "Attempting to pull image {}...", &source_image);
+
+        let mut pull_command = Command::new(SKOPEO_PATH);
+        pull_command
+            .arg("--insecure-policy")
```

Comment:
This needs a comment explaining why we are using an insecure option for Confidential Computing imho ;)

Reply:
I am not very clear about this. In my test, it can work without this option, and this option may be removed in the future.

Reply:
Having that option looks a bit alarming to the reader tbh :fearful:

Reply:
@jodh-intel you are right. It definitely appears alarming, especially in the context of confidential computing ;) . However, if you are ok, for next demo we can leave it as-is with a comment. 
Ideally if we stick to skopeo for image pull in the long run, then we need to create a proper policy file and use the same.  This was discussed briefly here as well - https://github.com/kata-containers/kata-containers/pull/2733/files#r740133328

Reference for policy: https://github.com/containers/skopeo/blob/main/docs/skopeo.1.md

Reply:
A comment is fine for now imho.

Reply:
added